core/modules/layout/layout.admin.inc

```diff
@@ -2653,75 +2654,86 @@ function layout_ajax_form_update($form, $form_state) {
  * Administrative form for layout settings.
  */
 function layout_settings_page($form, &$form_state) {
+  $header = array(
+    'layout_template_info' => array('data' => t('Template'), 'class' => array(RESPONSIVE_PRIORITY_HIGH)),
+    'type' => array('data' => t('Type'), 'sort' => 'desc', 'class' => array(RESPONSIVE_PRIORITY_LOW)),
+    'operations' => t('Operations'),
+  );
+
   backdrop_set_title(t('Layout settings'));
   $excluded = config_get('layout.settings', 'excluded_templates');
 
-  $form = array();
-
-  $form['#attached']['js'][] = backdrop_get_path('module', 'layout') . '/js/layout.admin.js';
-  $form['#attached']['css'][] = backdrop_get_path('module', 'layout') . '/css/layout.admin.css';
-  $form['#attributes']['class'] = array('layout-settings-page');
-
-  // Get the list of layout template options. The list needs to be rebuilt (see
-  // https://github.com/backdrop/backdrop-issues/issues/984)
-  $all_template_info = layout_get_layout_template_info(NULL, TRUE);
+  $rows = array();
   $options = array();
   $default = array();
   $form_state['hidden_templates'] = array();
+  $all_template_info = layout_get_layout_template_info(NULL, TRUE);
+
   foreach ($all_template_info as $template_name => $template_info) {
     // Build the default values list.
+    $is_default = FALSE;
     if (!in_array($template_name, $excluded)) {
-      $default[] = $template_name;
+      $is_default = $default[$template_name] = TRUE;
     }
 
     // Build the list of possible options, either enabled or not hidden.
-    if (in_array($template_name, $default) || !$template_info['hidden']) {
-      $options[$template_name] = theme('layout_template_option', array('template_info' => $template_info));
+    if ($is_default || !$template_info['hidden']) {
+      $rows[$template_name] = array(
+        'layout_template_info' => theme('layout_template_info', array('template_info' => $template_info)),
+        'type' => !empty($template_info['flexible']) ? t('Flexible') : t('Standard'),
+      );
+      $rows[$template_name]['operations']['data'] = array(
+        '#type' => 'dropbutton',
+        '#links' => _layout_settings_get_group_operations($template_info, $is_default),
+       );
     }
     // For disabled and hidden templates, save the fact they are disabled.
     elseif ($template_info['hidden']) {
       $form_state['hidden_templates'][] = $template_name;
     }
   }
 
-  $form['templates'] = array(
-    '#title' => t('Available layout templates'),
-    '#description' => t('Only templates that are selected here will be available when configuring layouts.'),
-    '#type' => 'checkboxes',
+  $form['#attached']['css'][] = backdrop_get_path('module', 'layout') . '/css/layout.admin.css';
+  $form['table'] = array(
+    '#type' => 'tableselect',
```

Comment:
I'm not sure tableselect is the right tool here? Would you ever want to disable all layout templates at the same time? This is sort of similar to the modules page, where a select-all checkbox might not make sense.

It's also not clear based on the current form what the checkbox even does. Maybe replacing the select-all checkbox with just a label for enabled would be more clear:

![image](https://user-images.githubusercontent.com/100206/62427263-a841e900-b6ad-11e9-8b9c-f854a686b676.png)

Reply:
Done. Converted to plain table and an enabled indicator.

Reply:
On the other hand, reverted this back to a tableselect but without the select all checkbox. We'd lose the ability to enable/disable multiple templates at once unless we had checkboxes. If its unclear might need a `#description`?